src/Table/TableHead.tsx

```diff
@@ -1,26 +1,32 @@
-// @ts-nocheck
 import React from "react";
 import styled from "styled-components";
 import StyledTh from "./StyledTh";
-import { Columns } from "./Table.types";
+import type { ColumnType, Columns } from "./Table.types";
 
-type TableHeadProps = {
+interface TableHeadProps {
   columns: Columns;
   compact?: boolean;
   sticky?: boolean;
-};
+}
 
 const StyledHeaderRow = styled.tr(({ theme }) => ({
   color: theme.colors.darkGrey,
   borderBottom: `1px solid ${theme.colors.lightGrey}`,
 }));
 
-const defaultheaderFormatter = ({ label }) => label;
-const renderHeaderCellContent = ({ headerFormatter = defaultheaderFormatter, ...column }) => headerFormatter(column);
+const defaultheaderFormatter: ColumnType["headerFormatter"] = ({ label }) => label;
+
+const renderHeaderCellContent = ({
+  headerFormatter = defaultheaderFormatter,
+  align,
+  label,
+  dataKey,
+  width,
+}: ColumnType) => headerFormatter({ align, label, dataKey, width });
```

Comment:
This is possibly a breaking change, but it complies with the type described in `Table.types.ts`.